packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2418,11 +2418,40 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     updateKeepAlive();
   }
 
+  String _cachedText = '';
+  double _cachedScrollOffset = 0.0;
+  Rect? _cachedFirstRect;
+
+  void _updateSelectionRects() {
+    if (defaultTargetPlatform != TargetPlatform.iOS)
+      return;
+    final TextSpan textSpan = buildTextSpan();
+    final StringBuffer text = StringBuffer();
+    textSpan.computeToPlainText(text);
+    final Rect firstRect = renderEditable.getBoxesForSelection(const TextSelection(baseOffset: 0, extentOffset: 1)).first.toRect();
+    final double scrollOffset = _scrollController?.offset ?? 0.0;
+    final ScrollDirection scrollDirection = _scrollController?.position.userScrollDirection ?? ScrollDirection.idle;
+    if (scrollDirection == ScrollDirection.idle && text.toString() != _cachedText ||
+        scrollOffset != _cachedScrollOffset ||
+        _cachedFirstRect != firstRect) {
+      _cachedText = text.toString();
+      _cachedScrollOffset = scrollOffset;
+      _cachedFirstRect = firstRect;
+      final Offset rectOffset = Offset(_isMultiline ? 0.0 : -1 * scrollOffset, _isMultiline ? -1 * scrollOffset : 0.0);
+      final List<Rect> rects = List<Rect>.generate(
+              text.length, (int i) => renderEditable.getBoxesForSelection(TextSelection(baseOffset: i, extentOffset: i + 1)).first.toRect())
```

Comment:
I'm not sure if this is going to work when there's surrogate pairs & extended glypheme clusters. @justinmc

Reply:
Will UIkit ever ask for the locations of the glyphs in a text field that is not focused? If that's the case the Rects can be outdated since we only update for the current text field.

Reply:
I believe it will not. When not focused, we should only be dealing with `UIIndirectScribbleInteraction` which asks for the elements in a rect (`requestElementsInRect`) and then lets us know to focus on one of them at a given point (`focusElementIfNeeded`). Once an element has focus, we should be sending over the rects so that UIKit has what it needs to do the text operations that happen during a `UIDirectScribbleInteraction`